service/history/historyEngine.go

```diff
@@ -154,6 +154,8 @@ var (
 	ErrConsistentQueryBufferExceeded = &workflow.InternalServiceError{Message: "consistent query buffer is full, cannot accept new consistent queries"}
 	// ErrConcurrentStartRequest is error indicating there is an outstanding start workflow request. The incoming request fails to acquires the lock before the outstanding request finishes.
 	ErrConcurrentStartRequest = &workflow.ServiceBusyError{Message: "an outstanding start workflow request is in-progress. Failed to acquire the resource."}
+	// ErrUnsupportedWorkflow is the error indicating cadence failed to process 2dc workflow type.
+	ErrUnsupportedWorkflow = &workflow.BadRequestError{Message: "versionHistories is empty, which is required for NDC feature. It's probably from deprecated 2dc workflows"}
```

Comment:
nit: probably define this error in execution package so that all places can use this pre-defined error?

Reply:
updated